lib/ansible/plugins/connection/ssh.py

```diff
@@ -309,8 +309,49 @@ def _examine_output(self, source, state, b_chunk, sudoable):
 
         return b''.join(output), remainder
 
+
     def _run(self, cmd, in_data, sudoable=True):
         '''
+        Will retry execution if ssh were unable to connect or timeout
+        '''
+        remaining_tries = int(C.ANSIBLE_SSH_RETRIES) + 1
+        cmd_summary = ' '.join(cmd)
+        for attempt in range(remaining_tries):
+            try:
+                return_tuple = self._run_once(cmd=cmd, in_data=in_data, sudoable=sudoable)
+                # 0 = success
+                # 1-254 = remote command return code
+                # 255 = failure from the ssh command itself
+                if return_tuple[0] != 255:
+                    break
+                else:
```

Comment:
Also pre-existing code, but I don't see any reason _run_once() shouldn't be responsible for raising AnsibleConnectionFailure on return_code = 255.

And then _run_once could raise AnsibleConnectionFailure errors with finer granularity (like AnsibleConnectionTimeout)
(https://github.com/ansible/ansible/pull/16649 attempts to do some of that, amongst other things)